src/python/pants/backend/native/tasks/link_shared_libraries.py

```diff
@@ -61,13 +61,11 @@ def implementation_version(cls):
 
   class LinkSharedLibrariesError(TaskError): pass
 
-  @memoized_property
-  def _cpp_toolchain(self):
-    return self._request_single(LLVMCppToolchain, self._native_toolchain).cpp_toolchain
-
   @memoized_property
   def linker(self):
-    return self._cpp_toolchain.cpp_linker
+    # NB: we are using the C++ toolchain here for linking every type of input, including compiled C
```

Comment:
Curious why this is necessary and why we don't link through the frontend of the C toolchain?

Reply:
Well, we would have to choose one, as in this phase we are linking C and C++ object files together into the shared lib defined by the `native_artifact()` build object. In this case, g++/clang++ are necessary to add the appropriate arguments to the linker subcommand for C++ (which can be seen if we add `-v` to the command line). This will also work for C. Because we are linking all these object files at once, we are forced to use the C++ compiler to support linking the C++ object files. We could e.g. use our C compiler to compile a target closure of only C targets, but I don't see any benefit to doing that. The reason we still have a specific C linker is so that we can support any C-specific link process in the future (and we do test our C linker in `test_native_toolchain.py`, so that's not going to rot).